### ISSUE TITLE
Analyzer performs override check between superinterfaces and superclass

### ISSUE THREAD

**Issue by askeksa-google** [dart-lang/sdk#34392]
```dart
abstract class I {
  foo([a]);
}

abstract class A {
  foo() {}
}

abstract class B extends A implements I {}

class C extends B {
  foo([a]) {}
}
```
The analyzer complains (at the declaration of `B`) that the `foo` method in `A` is not a valid override of the `foo` method in `I`.

There is no override relationship between those two methods, so having them both in `B` is valid, because all of the following hold:
- One of the methods has a signature which is a valid override of the other (so there is a most specific one for the interface of the class).
- The class is abstract, so it doesn't have to have implementations for all methods in its interface.
- Every concrete subclass has an implementation that is a valid override of both methods (so the method implementation satisfies the override rules and the class implements its interface properly).

This looks like it could be some mix-up between override checking and checking that every concrete class conforms to its interface.

**Comment by jmesserly** [comment]
@leafpetersen - did we do this on purpose?

IIRC, strong mode initially did not allow abstract members to override concrete ones with a different signature. But that may have been for simplicity of the implementation. We could certainly relax this and accept this example. (Not sure if there's a Dart 2 spec for how this should work?)

As I was thinking about this, it occurred to me that abstract member overrides could probably break soundness in Analyzer, and sure enough:
```dart
class C {
  Object foo() => 42;
}
abstract class D extends C {
  String foo([int x]);
}
class E extends D {}

main() {
  String s = E().foo(123); // OOPS!
  print(s.trim());
}
```
So we've got to fix something here regardless. If we fix how concrete classes are checked against their interfaces, both of these examples will behave correctly (accepting the initial example, rejecting my example above).

**Comment by askeksa-google** [comment]
I have filed the missing interface checks in the analyzer as https://github.com/dart-lang/sdk/issues/34507.

**Comment by scheglov** [comment]
OK, given that the specification is not complete, I'm moving this out of `Dart2.1` milestone for analyzer.

**Comment by eernstg** [comment]
Actually, dartLangSpec.tex _does_ specify an error, as of a07b2a10c6da58e0f3c461d778e51b128e2772a5, whenever there is no most-specific signature for a given member `m` (so the superinterfaces have multiple signatures for `m`, none of which is most specific, and the subclass itself does not declare `m`). I thought that change was part of a non-landed CL. But this means that there is no doubt that such a conflict is an error, and there is certainly no reason to consider this issue blocked on any spec updates.

**Comment by leafpetersen** [comment]
This would be good to fix for 2.1.

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/76061 (re)implements handling inheritance according to the updated spec, and how it was explained to me in mail. This fixes Analyzer for the original code, so it does not report any error.
